File: app/src/main/java/com/robinfinch/journal/app/WalkEntryActivity.java
package com.robinfinch.journal.app;

import android.app.Fragment;
import android.net.Uri;

import com.robinfinch.journal.app.ui.DetailsActivity;

/**
 * Walk entry details activity.
 *
 * @author Mark Hoogenboom
 */
public class WalkEntryActivity extends DetailsActivity implements WalkEntryFragment.Parent {

    @Override
    protected Fragment newFragmentFor(Uri uri) {
        return WalkEntryFragment.newInstance(uri);
    }

    @Override
    public void onWalkEntryDeleted() {
        finish();
    }
}

File: app/src/main/scala/com/robinfinch/journal/app/WalkEntryActivity.scala
package com.robinfinch.journal.app;

import android.net.Uri
import com.robinfinch.journal.app.ui.DetailsActivity

/**
 * Walk entry details activity.
 *
 * @author Mark Hoogenboom
 */
class WalkEntryActivity extends DetailsActivity with WalkEntryFragment.Parent {

    override def newFragmentFor(uri : Uri) = WalkEntryFragment.newInstance(uri)

    override def onWalkEntryDeleted() {
        finish();
    }
}
